### PR TITLE
omit empty span tags in the output

### PR DESCRIPTION
**Pull request by nknapp**

### CODE REVIEW

File: src/highlight.js [hunk 1]
```diff
@@ -418,6 +418,9 @@ https://highlightjs.org/
     }
 
     function buildSpan(classname, insideSpan, leaveOpen, noPrefix) {
+      if (!leaveOpen && insideSpan !== '') {
```
Comment (yyyc514) [inline src/highlight.js]: Wait wouldn't it be `insideSpan === ''`?  What am I missing?

Comment (nknapp) [reply src/highlight.js]: I made it wrong, changed it in the generated code, ran the tests and everything worked ;-)

**Comment by nknapp** [comment]
I moved the other if-statement to the top of the method because the rest of the method does not need to be executed if not className is set.

**Comment by yyyc514** [comment]
> I moved the other if-statement to the top of the method because the rest of the method does not need to be executed if not className is set.

Yeah I saw that the other day but wasn't worth a PR on it's own.  Good catch.

**Comment by yyyc514** [comment]
@egor-rogov Unless you see some reason we should leave these empty tags around this seems pretty clear cut to me.

**Comment by yyyc514** [comment]
@nknapp Can you update the changelog, please?

**Comment by nknapp** [comment]
#2175  as well?

**Comment by yyyc514** [comment]
Yes, we should always try to remember to update the changelog. :-)

**Pull request merged**